account_invoice_amount_adj/models/account_invoice.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2023 Quartile Limited
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class AccountInvoice(models.Model):
+    _inherit = "account.invoice"
+
+    @api.one
+    @api.depends(
+        "invoice_line_ids.price_subtotal",
+        "tax_line_ids.amount",
+        "tax_line_ids.amount_rounding",
+        "currency_id",
+        "company_id",
+        "date_invoice",
+        "type",
+    )
+    def _compute_amount(self):
+        super()._compute_amount()
+        # Use the original logic if a tax other than 
+        # the inclusive tax is applied to account.invoice.line.
+        if any(
+            not tax.price_include
+            for tax in self.invoice_line_ids.mapped("invoice_line_tax_ids")
+        ):
```

Comment:
Does your code work when there is no tax assignment?

```suggestion
        taxes = self.invoice_line_ids.mapped("invoice_line_tax_ids")
        if not taxes or not taxes.filtered(lambda x: x.price_include):
```

Please apply the same to the sales and purchase modules if this suggestion is appropriate.